src/lib/middleware/pat-middleware.ts

```diff
@@ -0,0 +1,38 @@
+import { IUnleashConfig } from '../types';
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+const patMiddleware = (
+    {
+        getLogger,
+        authentication,
+    }: Pick<IUnleashConfig, 'getLogger' | 'authentication' | 'flagResolver'>,
+    { userService }: any,
+): any => {
+    const logger = getLogger('/middleware/pat-middleware.ts');
+    logger.debug('Enabling PAT middleware');
+
+    if (!authentication.enablePersonalAccessToken) {
+        return (req, res, next) => next();
+    }
+
+    return async (req, res, next) => {
+        if (req.user) {
+            return next();
+        }
+
+        try {
+            const apiToken = req.header('authorization') as string;
```

Comment:
Same here. If we want this explicitly as string maybe it would make sense to type this in a different way? See e.g. `unleash-types.ts` - Maybe we could extend the headers in a similar way.
```suggestion
            const apiToken = req.header('authorization');
```

Reply:
Good one. Changed to correct types.